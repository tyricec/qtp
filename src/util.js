const util = {
  createElement: function(type) {
    const element = document.createElement(type)
    return element
  },
}

export default util