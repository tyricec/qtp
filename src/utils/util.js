const util = {
  buildHTML: (type, attrs, inner) => {
    let element

    if (type) {
      element = document.createElement(type)
    }
    
    if (element && attrs) {
      Object.keys(attrs).forEach((attr) => {
        element.setAttribute(attr, attrs[attr])
      })
    }

    if (element && inner) {
      element.appendChild(inner)
    }

    return element
  },
  buildText: (text) => {
    return document.createTextNode(text)
  },
}

export default util