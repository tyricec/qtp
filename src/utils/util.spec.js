import util from './util'

test('buildHTML creates div element', () => {
  const result = util.buildHTML('div')

  expect(result.outerHTML).toMatchSnapshot()
})

test('buildHTML create li element', () => {
  const result = util.buildHTML('li')

  expect(result.outerHTML).toMatchSnapshot()
})

test('buildHTML adds attribues', () => {
  const result = util.buildHTML(
    'li',
    {
      class: 'list-item',
      style: 'top: 20px;color: white;left: 5px',
    }
  )

  expect(result.outerHTML).toMatchSnapshot()
})

test('buildHTML adds text', () => {
  const result = util.buildHTML(
    'div',
    {},
    document.createTextNode('Build HTML Test')
  )

  expect(result.outerHTML).toMatchSnapshot()
})

test('buildText creates text node', () => {
  const result = util.buildText(
    'Build Text Test'
  )

  expect(result.wholeText).toMatchSnapshot()
})