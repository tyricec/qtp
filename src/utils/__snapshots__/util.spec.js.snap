// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`buildHTML adds attribues 1`] = `"<li class=\\"list-item\\" style=\\"top: 20px;color: white;left: 5px\\"></li>"`;

exports[`buildHTML adds text 1`] = `"<div>Build HTML Test</div>"`;

exports[`buildHTML create li element 1`] = `"<li></li>"`;

exports[`buildHTML creates div element 1`] = `"<div></div>"`;

exports[`buildText creates text node 1`] = `"Build Text Test"`;
